File: app/src/main/java/com/my/televip/features/other/FixTLError.java
package com.my.televip.features.other;

import com.my.televip.Class.ClassNames;
import com.my.televip.Class.ClassLoad;
import com.my.televip.Configs.ConfigManager;
import com.my.televip.base.BaseMethodHook;
import com.my.televip.hooks.HMethod;
import com.my.televip.logging.Logger;
import com.my.televip.obfuscate.ArgsResolver;
import com.my.televip.obfuscate.Obfuscate;
import com.my.televip.virtuals.messenger.NotificationCenter;

public class FixTLError {

    public static boolean isEnable = false;

    public static void init(){
        try {
            if (!isEnable) {
                isEnable = true;
                if (ClassLoad.getClass(ClassNames.LAUNCH_ACTIVITY) != null) {
                    HMethod.hookMethod(ClassLoad.getClass(ClassNames.LAUNCH_ACTIVITY), Obfuscate.getMethodName("LaunchActivity", "didReceivedNotification"), ArgsResolver.merge("didReceivedNotification", new Class[]{int.class, int.class, Object[].class}, new BaseMethodHook() {
                        @Override
                        protected void beforeMethod(MethodHookParam param) {
                            int id = (int) param.args[0];
                            if (id == NotificationCenter.getTlSchemeParseException() && ConfigManager.fixTLError.isEnable())
                                param.setResult(null);
                        }
                    }));
                }
            }
        } catch (Throwable t){
            Logger.e(t);
    }
    }

}

File: app/src/main/java/com/my/televip/features/other/FixTLError.kt
package com.my.televip.features.other

import com.my.televip.clazz.ClassLoad
import com.my.televip.clazz.ClassNames
import com.my.televip.configs.ConfigManager
import com.my.televip.base.BaseMethodHook
import com.my.televip.logging.Logger
import com.my.televip.obfuscate.ArgsResolver
import com.my.televip.obfuscate.Obfuscate
import com.my.televip.virtuals.messenger.NotificationCenter
import de.robv.android.xposed.XposedHelpers

object FixTLError {

    @JvmField
    var isEnable: Boolean = false

    @JvmStatic
    fun init() {
        try {
            if (!isEnable) {
                isEnable = true
                val launchClass = ClassLoad.getClass(ClassNames.LAUNCH_ACTIVITY)
                if (launchClass != null) {
                    val types: Array<Class<*>> = arrayOf(
                        Int::class.javaPrimitiveType!!,
                        Int::class.javaPrimitiveType!!,
                        Array<Any>::class.java
                    )
                    val merged = ArgsResolver.merge("didReceivedNotification", types, object : BaseMethodHook() {
                        override fun beforeMethod(param: MethodHookParam) {
                            val id = param.args[0] as Int
                            if (id == NotificationCenter.getTlSchemeParseException() && ConfigManager.fixTLError?.isEnable == true) {
                                param.result = null
                            }
                        }
                    })
                    if (merged != null) {
                        XposedHelpers.findAndHookMethod(
                            launchClass,
                            Obfuscate.getMethodName("LaunchActivity", "didReceivedNotification"),
                            *merged
                        )
                    }
                }
            }
        } catch (t: Throwable) {
            Logger.e(t)
        }
    }
}
